apps/kita/app/Http/Controllers/Admin/LoginController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use App\Providers\RouteServiceProvider;
+use Illuminate\Foundation\Auth\AuthenticatesUsers;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Http\Request;
+class LoginController extends Controller
+{
+    /*
+    |--------------------------------------------------------------------------
+    | Login Controller
+    |--------------------------------------------------------------------------
+    |
+    | This controller handles authenticating users for the application and
+    | redirecting them to your home screen. The controller uses a trait
+    | to conveniently provide its functionality to your applications.
+    |
+    */
+
+    use AuthenticatesUsers;
+
+    public function logout(Request $request)
+    {
+        \Auth::guard('admin_users')->logout();
+
+        $request->session()->regenerateToken();
+
+        if ($response = $this->loggedOut($request)) {
+            return $response;
+        }
+
+        return $request->wantsJson()
+            ? new JsonResponse([], 204)
+            : redirect('/');
+    }
```

Comment:
invalidateがsessionのstoreを全部消しちゃっている処理だったんですね。
そもそも今の実装だったら、logoutしてリダイレクトだけさせるんでいいようなきがするんでシンプルになるきがします。
regenerateTokenはloginのときにも走るので特に気にする必要内のかなって思ってたり。(あってもいいと思いますが)

```suggestion
    public function logout(Request $request)
    {
        $this->guard()->logout();
        
        return redirect()->route('showLoginForm');
    }
```

Reply:
指摘して頂いた通りに修正しました。
membersとadmin_usersのどちらでもログインしている時にはどっちのguardをログアウトするか指定しないといけないと思っていたのですが、$this->guard()->logout();だけできちんと動いていて驚きました。
なぜguardに指定をしなくても片方だけログアウトされるのかピンときていないのでお手隙の際に教えていただけると嬉しいです。

Reply:
あ、ごめんなさいコメント見てなかった。
```php
$this->guard()->logout();
```
この$this->guard()を見てみるとAuth::guard('admin_users')になって指定しているのですでに書いてあるものは使いましょう精神です。